src/main/java/org/apache/mesos/offer/DynamicPortRequirement.java

```diff
@@ -0,0 +1,95 @@
+package org.apache.mesos.offer;
+
+import org.apache.mesos.Protos;
+
+/**
+ * This class allows the expression of a desire for a random port designated by a name which will be injected into the
+ * environment.
+ */
+public class DynamicPortRequirement extends ResourceRequirement {
+    private static final String DYNAMIC_PORT_KEY = "dynamic_port_key";
```

Comment:
Does anything else store labels in resources at this point? Feels like they should all go in one place (ResourceUtils?), similar to how TaskUtils is accumulating TaskInfo-level labels.